command/agent/alloc_endpoint.go

```diff
@@ -586,33 +596,37 @@ func (s *HTTPServer) execStreamImpl(ws *websocket.Conn, args *cstructs.AllocExec
 		// don't close ws - wait to drain messages
 	}()
 
-	// Create a channel that decodes the results
-	errCh := make(chan HTTPCodedError, 2)
+	// Create a channel for the final result
+	resultCh := make(chan HTTPCodedError)
```

Comment:
This needs to be buffered to prevent L610 from blocking and leaking the goroutine if this func exits first.

Reply:
My first thought here is that there aren't any code paths out of this function that don't involve polling on the `resultCh` first. But you're right that a panic in the `handler` function could cause the outer (request-scoped) goroutine to exit, which would get recovered by the HTTP server.

(I think the original reason it was buffered was because both the goroutine below and the `forwardExecInput` both can send on the channel but we'd only ever pull one result in this request-scoped goroutine.)